manifests/init.pp

```diff
@@ -50,9 +50,15 @@
 
   if has_key($java::params::java, $distribution) {
     $default_package_name     = $java::params::java[$distribution]['package']
-    $default_alternative      = $java::params::java[$distribution]['alternative']
-    $default_alternative_path = $java::params::java[$distribution]['alternative_path']
-    $java_home                = $java::params::java[$distribution]['java_home']
+    if has_key($java::params::java[$distribution], 'alternative') {
+      $default_alternative = $java::params::java[$distribution]['alternative']
```

Comment:
Shall I remove it then ? If some other class needs it it should get it from `java::params`, shouldn't ?